pandas/core/frame.py

```diff
@@ -1750,10 +1750,13 @@ def to_csv(self, path_or_buf=None, sep=",", na_rep='', float_format=None,
         encoding : string, optional
             A string representing the encoding to use in the output file,
             defaults to 'ascii' on Python 2 and 'utf-8' on Python 3.
-        compression : {'infer', 'gzip', 'bz2', 'zip', 'xz', None}, default None
+        compression : {'infer', 'gzip', 'bz2', 'zip', 'xz', None},
+                      default 'infer'
             If 'infer' and `path_or_buf` is path-like, then detect compression
             from the following extensions: '.gz', '.bz2', '.zip' or '.xz'
             (otherwise no compression).
+            .. versionchanged:: 0.24.0
```

Comment:
Done in https://github.com/pandas-dev/pandas/pull/22011/commits/f8829a60a24a17451134014da27bd5f1f93a3fea, but would be good to hear from @TomAugspurger and @datapythonista, since we have complex situations such as:

```rst
DOCLINE
DOCLINE
.. versionchanged:: 0.23.0
   here is what was added
.. versionchanged:: 0.24.0 here is what changed

DOCLINE
```

For example, is the above OKAY or do we need additional blanks?

Reply:
@dhimmel I think you need the additional blank lines (before, and not sure if after).

The reason is not that much an standard in this case, but about sphinx understanding the directive. What we expect in the documentation, is that it's rendered like in the `validate` method of the `merge` docstring: https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.merge.html

But if you don't leave the right blank lines, sphinx doesn't detect it's a directive, and the text is rendered as it is. See this case: https://pandas.pydata.org/pandas-docs/version/0.23.1/generated/pandas.IntervalIndex.from_tuples.html

So, the best is if you can build the documentation, and check that it's rendered all right. This can be done by `./doc/make.py html` (or `./doc/make.py html --single pandas.DataFrame.read_csv`)

Let me know if you have any issue.

Reply:
Updated in https://github.com/pandas-dev/pandas/pull/22011/commits/eadf68e4e23c4d6715347d3096e4b50e99baaa0e and built the docs locally to confirm they're rendering properly.

Turns out the blank line before is required. After is not required. In between multiple statements is not required.